### ISSUE TITLE
Initializing multiple Actors freezes engine

### ISSUE THREAD

**Issue by natolambert** [huggingface/simulate#341]
I'm playing with creating multiple actors for multi-agent, and it seems like the naming of Camera's is causing some problems.

This snipped
```
    init_pos1 = np.random.random(3)
    actor1 = sm.EgocentricCameraActor(
        name="actor1", position=init_pos1, camera_tag="CameraSensor1"
    )  # Has a collider by default

    init_pos2 = np.random.random(3)
    actor2 = sm.EgocentricCameraActor(
        name="actor2", position=init_pos2, camera_tag="CameraSensor2"
    )  # Has a collider by default

    init_pos3 = np.random.random(3)
    actor3 = sm.EgocentricCameraActor(
        name="actor3", position=init_pos3, camera_tag="CameraSensor3"
    )  # Has a collider by default
```

Causes the engine to hang. Can we make it so the sensor names are procedurally generated? When they are all the same, the observations get merged and it makes the camera data grow with number of actors which is counter intutitive.

**Comment by natolambert** [comment]
Update, the camera dimension thing may be the number of actors impacting the returned data by repeating it (which could be intended), I am investigating. 

The hanging still is important.

**Comment by natolambert** [comment]
Update, Update: This may be due to data sent from the backend to the python API when increasing number of actors. Timeout is not clear.

**Comment by natolambert** [comment]
This will be resolved as #342 is developed most likely.
